File: FRONTEND/quickresto/src/Services/fournisseurservice.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/fournisseur";


export const createFournisseur = (data) => {
  return axios.post(`${API_URL}/createfournisseur`, null, {
    params: {
      nomFournisseur: data.nomFournisseur,
      quantite: data.quantite,
      unite: data.unite,
      statut: data.statut,
      ingredient: data.ingredient,
      prixUnitaire: data.prixUnitaire,
    },
  });
};



export const getAllFournisseurs = () => {
  return axios.get(`${API_URL}/allfournisseur`);
};
export const gettotalfournisseur=()=>{
   return  axios.get(`${API_URL}/totalfournisseur`);
}

export const deleteFournisseur = (id) => {
  return axios.delete(`${API_URL}/deletefournisseur/${id}`);
};

File: BACKEND/demo/src/main/java/com/example/demo/Service/FournisseurService.java
package com.example.demo.Service;

import org.springframework.stereotype.Service;
import com.example.demo.Modele.Fournisseur;
import com.example.demo.Repository.Fournisseurrepository;

import java.util.List;

@Service
public class FournisseurService {

    private final Fournisseurrepository fournisseurrepository;

    public FournisseurService(Fournisseurrepository fournisseurrepository) {
        this.fournisseurrepository = fournisseurrepository;
    }


    public String createFournisseur(String nomFournisseur, Integer quantite, String unite,
                                    String statut, String ingredient, Double prixUnitaire) {

        Fournisseur fournisseur = Fournisseur.builder()
                .nomFournisseur(nomFournisseur)
                .quantite(quantite)
                .unite(unite)
                .statut(statut)
                .ingredient(ingredient)
                .prixUnitaire(prixUnitaire)
                .build();

        fournisseurrepository.save(fournisseur);

        return "Fournisseur ajouté avec succès";
    }


    public List<Fournisseur> getAllFournisseurs() {
        return fournisseurrepository.findAll();
    }


    public String deleteFournisseur(Long id) {
        if (fournisseurrepository.existsById(id)) {
            fournisseurrepository.deleteById(id);
            return "Fournisseur supprimé avec succès";
        } else {
            return "Fournisseur introuvable";
        }
    }
    public long gettotalfournisseur(){
        return  fournisseurrepository.count();
    }
}